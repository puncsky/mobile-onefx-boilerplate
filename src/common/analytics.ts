import { Analytics, Event, PageHit } from "expo-analytics";
// @ts-ignore
import ExpoMixpanelAnalytics from "expo-mixpanel-analytics";

const ga = new Analytics("UA-43072488-4", undefined, { debug: __DEV__ });

const mixpanel = new ExpoMixpanelAnalytics("f3b829bced93a826d535abd48077fb28");

const analytics = {
  identify(id: string): void {
    mixpanel.identify(id);
    // @ts-ignore
    ga.parameters.uid = id;
  },

  // tslint:disable-next-line:no-any
  async track(name: string, props: Record<string, any>): Promise<void> {
    mixpanel.track(name, props);
    if (name.startsWith("page_view_")) {
      await ga.hit(new PageHit(name));
    } else if (name.startsWith("tap_")) {
      await ga.event(new Event("tap", name, props.id));
    } else {
      await ga.event(new Event(name, name));
    }
  },

  people_delete_user(): void {
    mixpanel.people_delete_user();
  }
};

export { analytics };
