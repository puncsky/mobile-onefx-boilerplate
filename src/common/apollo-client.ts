import { InMemoryCache } from "apollo-cache-inmemory";
import { ApolloClient } from "apollo-client";
import { ApolloLink } from "apollo-link";
import { HttpLink } from "apollo-link-http";
import fetch from "isomorphic-unfetch";
import { store } from "./store";

export const ENDPOINT = "https://guigu.io/";

export const headers: { [key: string]: string } = {
  "x-app-id": "mobile-guiguio"
};

export const getEndpoint = (path: string) => `${ENDPOINT}${path}`;

const middlewareLink = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: { authorization: `Bearer ${store.getState().base.authToken}` }
  });
  return forward(operation);
});

// use with apollo-client
const link = middlewareLink.concat(
  new HttpLink({
    uri: getEndpoint("api-gateway/"),
    fetch
  })
);

export const apolloClient = new ApolloClient({
  link,
  cache: new InMemoryCache()
});
