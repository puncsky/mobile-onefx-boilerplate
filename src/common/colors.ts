export const colors = {
  primary: "#FBB03B",
  secondary: "#0C8DE4",
  white: "#fff",

  black: "#000000",
  black90: "#333333",
  black80: "#999999",
  black60: "#CCCCCC",
  black40: "#E5E5E5",
  black20: "#F0F0F0",
  black10: "#F7F7F7",

  text01: "#4c4c4c", //		Primary text, Body copy

  error: "#E54937", //	Error
  success: "#07A35A", //	Success
  warning: "#FFA000", //	Warning
  information: "#5aaafa", //	Information

  nav01: "#011627", //	Global top bar
  nav02: "#20232a", //	CTA footer

  tabIconDefault: "#ccc",
  tabIconSelected: "#2f95dc"
};

export const antdTheme = {
  brand_primary: colors.primary,
  color_link: colors.primary,
  primary_button_fill: colors.primary,
  primary_button_fill_tap: colors.primary
};
