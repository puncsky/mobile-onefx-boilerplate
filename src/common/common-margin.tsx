import { View } from "react-native";
import React from "react";

export function CommonMargin(): JSX.Element {
  return <View style={{ margin: 8 }} />;
}
