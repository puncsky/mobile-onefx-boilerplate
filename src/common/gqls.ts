import gql from "graphql-tag";

export const ADD_PUSH_TOKEN = gql`
  mutation addPushToken($pushToken: String) {
    addPushToken(token: $pushToken)
  }
`;
