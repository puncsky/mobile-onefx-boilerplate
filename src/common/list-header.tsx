import React from "react";
import { Text } from "react-native";
import { ktheme } from "./theme";

export function ListHeader({ children }: { children: string }): JSX.Element {
  return (
    <Text
      style={{
        backgroundColor: ktheme.white,
        paddingLeft: 14,
        paddingRight: 14,
        paddingTop: 14,
        paddingBottom: 8,
        color: ktheme.text01
      }}
    >
      {children}
    </Text>
  );
}
