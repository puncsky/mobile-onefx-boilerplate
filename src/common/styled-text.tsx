// import * as React from "react";
// import { StyleProp, Text, TextStyle } from "react-native";

// interface Props {
//   style?: StyleProp<TextStyle>;
//   children: string;
// }

// export function MonoText(props: Props): JSX.Element {
//   const { children } = props;
//   return (
//     <Text {...props} style={[props.style, { fontFamily: "space-mono" }]}>
//       {children}
//     </Text>
//   );
// }
import * as React from "react";

import { Text, TextProps } from "./themed-text";

export function MonoText(props: TextProps) {
  return (
    <Text {...props} style={[props.style, { fontFamily: "space-mono" }]} />
  );
}
