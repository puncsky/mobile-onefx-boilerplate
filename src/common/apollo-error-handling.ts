import { onError } from "apollo-link-error";
import { store } from "@/common/store";

export const onErrorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    for (const err of graphQLErrors) {
      if (err.extensions && err.extensions.code === "UNAUTHENTICATED") {
        store.dispatch({
          type: "UPDATE_REDUX_STATE",
          payload: {
            base: {
              userId: "",
              authToken: ""
            }
          }
        });
      }
    }
  }
  if (networkError) {
    // eslint-disable-next-line no-console
    console.log(`[Network error]: ${networkError}`);
  }
});
