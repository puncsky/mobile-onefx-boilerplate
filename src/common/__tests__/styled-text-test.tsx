import * as React from "react";
import * as renderer from "react-test-renderer";
import { MonoText } from "@/common/styled-text";

it("renders correctly", () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();

  expect(tree).toMatchSnapshot();
});
