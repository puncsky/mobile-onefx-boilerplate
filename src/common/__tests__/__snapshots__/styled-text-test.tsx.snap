// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4c4c4c",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "space-mono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
